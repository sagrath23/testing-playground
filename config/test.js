const config = {
  pokeApiBaseUrl: 'https://a-very-fancy-url-to-get-pokemons'
};

module.exports = config;
