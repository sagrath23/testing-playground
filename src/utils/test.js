import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

const mockStore = (state) => {
  const defaultState = {
    detailedPokemon: {},
    currentPokemonList: {
      results: []
    },
    ...state
  };

  return {
    getState: jest.fn().mockReturnValue(defaultState),
    subscribe: jest.fn(),
    dispatch: jest.fn()
  };
};

const renderComponent = (store, component, entries) => render(
  <Provider store={store}>
    <MemoryRouter initialEntries={entries}>
      {component}
    </MemoryRouter>
  </Provider>
);

export {
  mockStore,
  renderComponent
};
