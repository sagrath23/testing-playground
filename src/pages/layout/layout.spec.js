import React from 'react';
import { mockStore, renderComponent} from '../../utils/test';
import Layout from './layout';

describe('Layout page Component test', () => {
  test('should render the component with default props', () => {
    const { container } = renderComponent(mockStore(), <Layout />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render the PokemonList Component', () => {
    const { container } = renderComponent(mockStore(), <Layout />, [{ pathname: '/pokemons', key: 'pokemons' }]);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render the PokemonDetail Component', () => {
    const { container } = renderComponent(mockStore(), <Layout />, [{ pathname: '/pokemons/pikachu', key: 'pokemons' }]);

    expect(container.firstChild).toMatchSnapshot();
  });
});
