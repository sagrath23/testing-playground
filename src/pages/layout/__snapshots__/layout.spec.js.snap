// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout page Component test should render the PokemonDetail Component 1`] = `
<nav
  class="navbar navbar-expand-md navbar-light bg-light"
>
  <a
    class="navbar-brand"
    href="/"
  >
    Awesome Pokedex
  </a>
  <button
    aria-label="Toggle navigation"
    class="navbar-toggler"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded="false"
    class="collapse navbar-collapse"
  >
    <ul
      class="ml-auto navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/components/"
        >
          Components
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="https://github.com/reactstrap/reactstrap"
        >
          GitHub
        </a>
      </li>
      <li
        class="dropdown nav-item"
      >
        <a
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle nav-link"
          href="#"
        >
          Options
        </a>
        <div
          aria-hidden="true"
          class="dropdown-menu dropdown-menu-right"
          role="menu"
          tabindex="-1"
        >
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Option 1
          </button>
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Option 2
          </button>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Reset
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Layout page Component test should render the PokemonList Component 1`] = `
<nav
  class="navbar navbar-expand-md navbar-light bg-light"
>
  <a
    class="navbar-brand"
    href="/"
  >
    Awesome Pokedex
  </a>
  <button
    aria-label="Toggle navigation"
    class="navbar-toggler"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded="false"
    class="collapse navbar-collapse"
  >
    <ul
      class="ml-auto navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/components/"
        >
          Components
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="https://github.com/reactstrap/reactstrap"
        >
          GitHub
        </a>
      </li>
      <li
        class="dropdown nav-item"
      >
        <a
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle nav-link"
          href="#"
        >
          Options
        </a>
        <div
          aria-hidden="true"
          class="dropdown-menu dropdown-menu-right"
          role="menu"
          tabindex="-1"
        >
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Option 1
          </button>
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Option 2
          </button>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Reset
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Layout page Component test should render the component with default props 1`] = `
<nav
  class="navbar navbar-expand-md navbar-light bg-light"
>
  <a
    class="navbar-brand"
    href="/"
  >
    Awesome Pokedex
  </a>
  <button
    aria-label="Toggle navigation"
    class="navbar-toggler"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded="false"
    class="collapse navbar-collapse"
  >
    <ul
      class="ml-auto navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/components/"
        >
          Components
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="https://github.com/reactstrap/reactstrap"
        >
          GitHub
        </a>
      </li>
      <li
        class="dropdown nav-item"
      >
        <a
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle nav-link"
          href="#"
        >
          Options
        </a>
        <div
          aria-hidden="true"
          class="dropdown-menu dropdown-menu-right"
          role="menu"
          tabindex="-1"
        >
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Option 1
          </button>
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Option 2
          </button>
          <div
            class="dropdown-divider"
            tabindex="-1"
          />
          <button
            class="dropdown-item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            Reset
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;
